android/src/main/java/com/swmansion/reanimated/nodes/ParamNode.java

```diff
@@ -17,31 +19,40 @@ public ParamNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
 
   @Override
   public void setValue(Object value) {
-    Node node = mNodesManager.findNodeById(mArgsStack.peek(), Node.class);
-    String callID = mUpdateContext.callID;
-    mUpdateContext.callID = mPrevCallID;
-    ((ValueNode) node).setValue(value);
-    mUpdateContext.callID = callID;
+    try {
+      Node node = mNodesManager.findNodeById(mArgsStack.peek(), Node.class);
+      String callID = mUpdateContext.callID;
+      mUpdateContext.callID = mPrevCallID;
+      ((ValueNode) node).setValue(value);
+      mUpdateContext.callID = callID;
+    } catch (EmptyStackException e) {
+      throw new JSApplicationCausedNativeException(getClass().getSimpleName() + " is trying to set value with no context.", e);
+    }
   }
 
+  @Override
   public void beginContext(Integer ref, String prevCallID) {
     mPrevCallID = prevCallID;
     mArgsStack.push(ref);
   }
 
-
+  @Override
   public void endContext() {
     mArgsStack.pop();
   }
 
 
   @Override
   protected Object evaluate() {
-    String callID = mUpdateContext.callID;
-    mUpdateContext.callID = mPrevCallID;
-    Node node = mNodesManager.findNodeById(mArgsStack.peek(), Node.class);
-    Object val = node.value();
-    mUpdateContext.callID = callID;
-    return val;
+    try {
+      String callID = mUpdateContext.callID;
+      mUpdateContext.callID = mPrevCallID;
+      Node node = mNodesManager.findNodeById(mArgsStack.peek(), Node.class);
+      Object val = node.value();
+      mUpdateContext.callID = callID;
+      return val;
+    } catch (EmptyStackException e) {
+      throw new JSApplicationCausedNativeException(getClass().getSimpleName() + " is trying to evaluate with no context.", e);
```

Comment:
It should never happen

Reply:
It happens with callback because it's async

Reply:
So the error should say so

Reply:
5d23b28